src/OrchardCore.Modules/OrchardCore.Markdown/Drivers/MarkdownFieldDriver.cs

```diff
@@ -38,7 +41,7 @@ public override IDisplayResult Display(MarkdownField field, BuildFieldDisplayCon
                 model.PartFieldDefinition = context.PartFieldDefinition;
                 templateContext.SetValue("Model", model);
 
-                model.Markdown = await _liquidTemplatemanager.RenderAsync(field.Markdown, System.Text.Encodings.Web.HtmlEncoder.Default, templateContext);
+                model.Markdown = await _liquidTemplatemanager.RenderAsync(field.Markdown, NullEncoder.Default, templateContext);
                 model.Html = Markdig.Markdown.ToHtml(model.Markdown ?? "");
```

Comment:
Idem, looks right to use the `NullEncoder` to initialize the `Markdown` field

But then, to set the `Html` field maybe we need

    model.Html = _htmlEncoder.Encode(Markdig.Markdown.ToHtml(model.Markdown ?? "");

Knowing that the related shape do `@Html.Raw(Model.Html)` <= but maybe wrong here

Note: For the html field shape we also uses `@Html.Raw(Model.Html)` and the driver doesn't use `_htmlEncoder`, so maybe we also need to update the html field driver.

Reply:
Liquid should use the html encoder here. And we should not encode the markdown.